src/rules/custom-property-no-empty-line-before/index.js

```diff
@@ -0,0 +1,62 @@
+import {
+  isCustomProperty,
+  optionsHaveException,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import _ from "lodash"
+
+export const ruleName = "custom-property-no-empty-line-before"
+
+export const messages = ruleMessages(ruleName, {
+  expected: "Expected single line before custom property",
+  rejected: "Unexpected single line before custom property",
+})
+
+export default function (expectation, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        true,
+      ],
+    }, {
+      actual: options,
+      possible: {
+        except: ["first-after-non-custom-property-sibling"],
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+    const allPropData = []
```

Comment:
Can you use `node.prev()` instead of storing properties this way? Also, won't this cause problems by storing properties from all blocks, not just the one the custom property is within?